### PR TITLE
feat(checkbox): support leading visuals

### PR DESCRIPTION
**Pull request by iplesko**
### Description

Allow clients to compose a checkbox with a decorative leading visual between the checkbox control and its existing label and help text.

An example of checkbox with the leading visual:

```html
<mdc-checkbox 
      label="Alex Example" 
      help-text="example.com" 
      toggletip-text="This participant is connected from a video device."
      info-icon-aria-label="About Alex Example">
  <mdc-avatar slot="leading-visual" size="32" aria-hidden="true"></mdc-avatar>
</mdc-checkbox>
```

Checkbox with the leading visual

### Breaking Change

None.

### CODE REVIEW

File: packages/components/src/components/checkbox/checkbox.stories.ts [hunk 1]
```diff
@@ -149,6 +159,33 @@ export const Indeterminate: StoryObj = {
...
+        size="32"
+        src=${imageFixtures.avatar}
+        presence=${PRESENCE_TYPE.ON_DEVICE}
+      ></mdc-avatar>
+      <span slot="label">Alex Example</span>
+      <span slot="supporting-text">example.com</span>
+      <mdc-button
+        slot="toggletip"
+        id="lobby-participant-info-trigger"
+        prefix-icon="info-circle-regular"
```
Comment (cscheffauer) [inline packages/components/src/components/checkbox/checkbox.stories.ts]: info-circle-filled is what is normally used and recommended - we should use this version here too.

File: packages/components/src/components/checkbox/checkbox.e2e-test.ts [hunk 2]
```diff
@@ -312,6 +379,64 @@ test('mdc-checkbox', async ({ componentsPage }) => {
...
+    });
+
+    await test.step('toggletip trigger should not toggle the checkbox', async () => {
+      await componentsPage.mount({
+        html: `
+          <mdc-checkbox label="Alex Example" supporting-text="example.com">
+            <mdc-button
+              slot="toggletip"
+              id="participant-info-trigger"
+              prefix-icon="info-circle-regular"
```
Comment (cscheffauer) [inline packages/components/src/components/checkbox/checkbox.e2e-test.ts]: info-circle-filled is what is normally used and recommended - we should use this version here too.

File: packages/components/src/components/checkbox/checkbox.e2e-test.ts [hunk 3]
```diff
@@ -210,6 +251,18 @@ test('mdc-checkbox', async ({ componentsPage }) => {
       await expect(label).toHaveText('Checkbox label');
     });
 
+    await test.step('should include supporting text in the accessible name', async () => {
+      await componentsPage.setAttributes(checkbox, {
+        label: 'Kristine Stone',
+        'supporting-text': 'cisco.com',
```
Comment (cscheffauer) [inline packages/components/src/components/checkbox/checkbox.e2e-test.ts]: lets use example.com everywhere

**Review by cscheffauer — CHANGES_REQUESTED** [review]
first round of review

File: packages/components/src/components/checkbox/checkbox.stories.ts (hunk 1 above)
Comment (iplesko) [reply packages/components/src/components/checkbox/checkbox.stories.ts]: Instead, I completely remvoed the toggletip slot usage. Instead, the examples use the checkbox's `toggletip-text` parameter as all the other tests or stories. That seems like more standard usage.

File: packages/components/src/components/checkbox/checkbox.e2e-test.ts (hunk 2 above)
Comment (iplesko) [reply packages/components/src/components/checkbox/checkbox.e2e-test.ts]: Same as above.

File: packages/components/src/components/checkbox/checkbox.e2e-test.ts (hunk 3 above)
Comment (iplesko) [reply packages/components/src/components/checkbox/checkbox.e2e-test.ts]: Fixed and also improved `AGENTS.md` to improve chance this won't happen again.

File: packages/components/src/components/checkbox/checkbox.e2e-test.ts [hunk 4]
```diff
@@ -186,11 +209,87 @@ test('mdc-checkbox', async ({ componentsPage }) => {
     });
   });
 
+  /**
+   * LAYOUT
+   */
+  await test.step('layout', async () => {
+    await test.step('should align a rich-label checkbox and preserve text-container box customizations', async () => {
```
Comment (iplesko) [inline packages/components/src/components/checkbox/checkbox.e2e-test.ts]: This was a real regression I introduced during development. It's quite hard to catch, imo, so I thought this test could be valuable. Let me know what you think, because on the other hand, the test is quite verbose, catches what is arguably an edge case. I would still keep it though.

File: packages/components/src/components/checkbox/checkbox.component.ts [hunk 5]
```diff
@@ -75,6 +90,17 @@ class Checkbox
    */
   @property({ type: Boolean, reflect: true }) indeterminate = false;
 
+  /**
+   * Supporting text that identifies or qualifies the checkbox option. It is part of the checkbox label target.
+   */
+  @property({ type: String, reflect: true, attribute: 'supporting-text' }) supportingText?: string;
```
Comment (cscheffauer) [inline packages/components/src/components/checkbox/checkbox.component.ts]: issue: help-text already exists in formfieldwrapper and should be used. Creating another one here is causing inconsistencies

**Review by cscheffauer — CHANGES_REQUESTED** [review]

File: packages/components/src/components/checkbox/checkbox.component.ts (hunk 5 above)
Comment (iplesko) [reply packages/components/src/components/checkbox/checkbox.component.ts]: Removed and simplified even further.

**Review by cscheffauer — APPROVED** [review]
